lib/puppet/face/module/build.rb

```diff
@@ -4,6 +4,7 @@
     description <<-EOT
       Prepares a local module for release on the Puppet Forge by building a
       ready-to-upload archive file.
+      Note: Module build is not supported/allowed on FIPS enabled systems.
```

Comment:
I'd use the more specific "prohibited" instead of "not supported/allowed" or "not supported", here and in the other docs and error message additions. I'd also explain why, instead of leaving the reason in the code comments.

So in each help section, I'd go with:

> Note: Module ... **uses MD5 checksums, which are prohibited on FIPS-enabled systems**.

And in the error messages, I'd go with:

> Module ... is **prohibited** in FIPS mode.

(changes in bold)